libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageViewModel.kt

```diff
@@ -3,27 +3,34 @@ package org.wordpress.android.imageeditor.preview
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageLoadToFileState.ImageLoadToFileFailedState
+import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageLoadToFileState.ImageLoadToFileIdleState
+import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageLoadToFileState.ImageLoadToFileSuccessState
+import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageLoadToFileState.ImageStartLoadingToFileState
 import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageUiState.ImageDataStartLoadingUiState
 import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageUiState.ImageInHighResLoadFailedUiState
 import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageUiState.ImageInHighResLoadSuccessUiState
 import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageUiState.ImageInLowResLoadFailedUiState
 import org.wordpress.android.imageeditor.preview.PreviewImageViewModel.ImageUiState.ImageInLowResLoadSuccessUiState
+import org.wordpress.android.imageeditor.viewmodel.Event
 
 class PreviewImageViewModel : ViewModel() {
-    private val _uiState: MutableLiveData<UiState> = MutableLiveData()
-    val uiState: LiveData<UiState> = _uiState
+    private val _uiState: MutableLiveData<Event<UiState>> = MutableLiveData()
+    val uiState: LiveData<Event<UiState>> = _uiState
```

Comment:
Can you please elaborate why we need to use Event for the UiState. 

`Event` is usually used for events we want to consume just once (eg. snackbar msgs, navigation) to make sure we don't perform the same navigation or show the same snackbar msg more than once (eg. after rotation).

Reply:
I noticed active observer count (`hasActiveObservers()`) increased every time for `uiState` `LiveData` on returning to preview from crop screen. To prevent excessive UI re-rendering I decided to use `Event`.

I didn't get a chance to investigate it but [Architecture Navigation Component : onCreateView gets called every time](https://stackoverflow.com/questions/54879581/architecture-navigation-component-oncreateview-gets-called-every-time) could be a potential reason. However I might be wrong. Let me know what you think. 

Quoting an [explanation](https://issuetracker.google.com/issues/109856764) from a linked issue: 

> Google doesn't want you to stack a bunch of fragments on top of each other using show/hide because it promotes bad app architecture and wastes available memory. .... Nobody is stopping you from keeping views in memory manually, but the whole point of these jetpack components is to stop you from doing these mistakes in the first place.

Reply:
Thanks for the clarification :)

>I noticed active observer count (hasActiveObservers()) increased every time for uiState LiveData on returning to preview from crop screen. To prevent excessive UI re-rendering I decided to use Event.

ohh, that'd mean we are leaking the fragment. Which is a high priority issue anyway, but I agree we can fix it after the project issues. ;)

Reply:
I'm not sure I follow the issue from the StackOverFlow. When a fragment is put on backstack we always want to invoke `onDestroyView` and whenever the fragment is poped from a backstack we always want to invoke `onCreateView`. This is imo expected behavior no matter if we use NavComp or not. As per the [fragment's lifecycle diagram](https://blog.avenuecode.com/hs-fs/hubfs/The%20activity%20and%20fragment%20lifecycles%20(1).png?width=640&name=The%20activity%20and%20fragment%20lifecycles%20(1).png).


I've checked `viewModel.uiState.mActiveCount` in the `setupObservers` method after I return back from the Crop fragment and the count is == 0 (even when we don't use `Event`). If we want to reduce the number of times the observer gets notified with a UI change, we might want to consider introducing a throttle on the uiState LiveData.  

Wdyt? Is there something I'm missing?

Reply:
I think it's safe to use the `Event`. It just something we don't usually do, but I don't see any downsides except of inconsistency. So feel free to keep it if you want. I'm glad that it seems we are not leaking the fragment :) - I don't like leaks:P.

Reply:
I really want to research bit more on this but later. Till that time let's keep `Event`.

Reply:
Added a [note](https://github.com/orgs/wordpress-mobile/projects/54#card-36418894) in the project to investigate it later.